### PR TITLE
fix: add net8 and net7 support

### PR DESCRIPTION
**Pull request by elonmallin**
Hi, I'd like to add .net8 and .net7 support to this great tool. This PR is the same as the last merged PR which added .net6.

What do you think about merging this when .net8 is released on 2023-11-14?

### CODE REVIEW

**Comment by elonmallin** [comment]
Hi @starcraft66 @Gcaya @asimmon @decarufe and sorry to tag you like this but this tool seems to be heavily used ([downloaded 700 times per day](https://www.nuget.org/packages/dotnet-certificate-tool)) even though it's unlisted. It also has 2 published spinoffs to add support for .NET7.

Is there any interest from Workleap to relist this nuget or give the ownership to me or someone else to support it?

**Comment by dradovic** [comment]
@elonmallin Thank you for your contribution. I did the .NET 7 upgrade last year. Unfortunately, nobody from maintainers of this repo ever replied so far.

Again, as I need it in my CI environment, I've pushed a version to nuget.org. Here's what I did:

1. Cloned your repo from `https://github.com/elonmallin/dotnet-certificate-tool.git`
2. Adjusted package ID to `<PackageId>dotnet-certificate-tool-net8</PackageId>` to not have a conflict on nuget.org
3. `dotnet build -c Release`
4. `dotnet pack -c Release GSoft.CertificateTool.csproj`
5. Uploaded to nuget.org

And here it is: https://www.nuget.org/packages/dotnet-certificate-tool-net8/1.0.0

**Comment by elonmallin** [comment]
Thanks @dradovic !

Are you interested in maintaining your version? Might be nice with a non framework specific name in that case and making a public repo for it :)

**Comment by dradovic** [comment]
I'm not necessarily keen to become a maintainer of this project as I'm not acquainted with its actual logic. I still hope that somebody of the original maintainers will step up or somebody else (you? ;).

But who knows. If we bump our heads against a .NET upgrade that requires more than just adjusting some .NET tags, we need a new maintained public repo.

**Comment by SoyDiego** [comment]
@gsoft-inc @yohanb will you approve this PR?

Thanks!

**Comment by SoyDiego** [comment]
@gsoft-inc @yohanb @patricklafrance @decarufe will you approve this PR?

Thanks!

**Review by yohanb — APPROVED** [review]

**Comment by stylianosnicoletti** [comment]
@elonmallin @yohanb @ivanpaulovich @shoshins can you please merge the PR

File: docker-example/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM mcr.microsoft.com/dotnet/sdk:6.0.100-bullseye-slim-amd64 as build-env
+FROM mcr.microsoft.com/dotnet/sdk:8.0.100-bullseye-slim-amd64 as build-env
```
Comment (stylianosnicoletti) [inline docker-example/Dockerfile:1]: use latest '8.0.201-bookworm-slim-amd64'

File: docker-example/Dockerfile [hunk 2]
```diff
@@ -13,7 +13,7 @@ RUN dotnet publish -c Release -o /app/out
 RUN dotnet tool install --tool-path /app/out dotnet-certificate-tool
 
 # Build runtime image
-FROM mcr.microsoft.com/dotnet/aspnet:6.0.0-bullseye-slim-amd64
+FROM mcr.microsoft.com/dotnet/aspnet:8.0.0-bullseye-slim-amd64
```
Comment (stylianosnicoletti) [inline docker-example/Dockerfile:16]: use latest '8.0.201-bookworm-slim-amd64'

**Review by stylianosnicoletti — CHANGES_REQUESTED** [review]

**Comment by stylianosnicoletti** [comment]
Hi @elonmallin the docker image used in your PR seems to be wrong/outdated. Can you please change it to the one suggested above?

**Comment by williamiss** [comment]
This is a widely used tool, are there plans to merge this PR?

**Comment by daphneSot** [comment]
Hello, can you merge this PR as we need it to upgrade our environments?

**Comment by decarufe** [comment]
See: [Seeking New Maintainers for dotnet-certificat-tool](https://github.com/gsoft-inc/dotnet-certificate-tool/issues/34)

**Review by stylianosnicoletti — APPROVED** [review]

**Pull request merged**